pkg/plugin/processor/builtin/convertField.go

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=convertField_paramgen.go convertFieldConfig
+
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type convertField struct {
+	referenceResolver sdk.ReferenceResolver
+	config            convertFieldConfig
+
+	sdk.UnimplementedProcessor
+}
+
+func newConvertField() *convertField {
+	return &convertField{}
+}
+
+type convertFieldConfig struct {
+	// Field is the target field, as it would be addressed in a Go template (e.g. `.Payload.After.foo`).
+	// you can only convert fields that are under .Key and .Payload, and said fields should contain structured data.
+	Field string `json:"field" validate:"required,regex=^\\.(Payload|Key)\\.*"`
```

Comment:
Umm we shouldn't escape the last `.` though, that actually represents "any character" 🙈